// food-ordering-backend/config/db.js
import mysql from 'mysql2/promise';

const createDatabaseConfig = () => {
  const databaseUrlValue = process.env.DATABASE_URL || process.env.MYSQL_URL || process.env.MYSQL_PUBLIC_URL;
  const databaseUrl = databaseUrlValue ? new URL(databaseUrlValue) : null;

  if (databaseUrl) {
    return {
      host: databaseUrl.hostname,
      port: Number(databaseUrl.port || 3306),
      user: decodeURIComponent(databaseUrl.username || 'root'),
      password: decodeURIComponent(databaseUrl.password || ''),
      database: databaseUrl.pathname.replace(/^\//, '') || 'railway',
    };
  }

  return {
    host: process.env.DB_HOST || process.env.MYSQLHOST || 'localhost',
    port: Number(process.env.DB_PORT || process.env.MYSQLPORT || 3306),
    user: process.env.DB_USER || process.env.MYSQLUSER || 'root',
    password: process.env.DB_PASSWORD || process.env.MYSQLPASSWORD || '',
    database: process.env.DB_NAME || process.env.MYSQLDATABASE || process.env.MYSQL_DATABASE || 'cravingdash',
  };
};

export const dbConfig = createDatabaseConfig();

export const pool = mysql.createPool({
  ...dbConfig,
  waitForConnections: true,
  connectionLimit: 10,
});
